### ISSUE TITLE
Boolean Literal types aren't considered for reachability

### ISSUE THREAD

**Issue by mauvealerts** [microsoft/pyright#3085]
**Describe the bug**
`Literal[True]` and `Literal[False]` aren't considered when determining whether code is reachable.  This manifests in several ways, including spurious errors and code not grayed out in VS Code.

**To Reproduce**
Run pyright against code below
```
  C:\Users\Mauve\dev\demo.py:10:7 - error: "x" is possibly unbound (reportUnboundVariable)
  C:\Users\Mauve\dev\demo.py:15:7 - error: "x" is possibly unbound (reportUnboundVariable)
2 errors, 0 warnings, 0 informations
```

**Expected behavior**
```
0 errors, 0 warnings, 0 informations
```

**Screenshots or Code**
```py
from typing import Literal


def foo() -> Literal[True]:
    return True

if foo():
    x = 1
print(x) # erroneous "x" is possibly unbound

b = True
if b:
    x = 2
print(x) # erroneous "x" is possibly unbound

if True:
    x = 3
print(x) # ok, as expected
```

**VS Code extension or command-line**
CLI, pyright 1.1.222

**Additional context**
The example is silly, but I encountered something like this while reviewing some code.

(For searchability: unreachable, dead)

**Comment by erictraut** [comment]
This is currently "by design". Pyright's logic does consider certain static expressions such as `True` when building its code flow graph, but it does not take into account type analysis for reachability. Doing so would be very costly from an analysis perspective. The recommended workaround is to not leave variables uninitialized on some code paths.
